sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_options.hpp

```diff
@@ -158,6 +158,13 @@ namespace Azure { namespace Storage { namespace Blobs {
      * API version used by this client.
      */
     std::string ApiVersion;
+
+    /**
+     * Enables tenant discovery through the authorization challenge when the client is configured to
+     * use a TokenCredential. When enabled, the client will attempt an initial un-authorized request
+     * to prompt a challenge in order to discover the correct tenant for the resource.
+     */
+    bool EnableTenantDiscovery = false;
```

Comment:
If you want, I think it is ok to not have this option (or delay having it) - effectively, `"EnableTenantDiscovery"` will be `"true"` when the credential is going to have something in the credential options'  `AdditionallyAllowedTenants` vector - additional tenant IDs, or a `"*"` value.
If it is empty (and it is empty by default), then the policy will effectively work like `BearerTokenAuthenticationPolicy` does today - in either case, even if you parse the TenantId from the challenge and set that, the credential will not let you to authorize via another TenantId, unless it is in its `options.AdditionallyAllowedTenants`.

Reply:
Otherwise, it think it will be more consistent to return `false` from `AuthorizeRequestOnChallenge()` right away if `EnableTenantDiscovery` is `false`. But up to you.

Reply:
Yeah, I think it's make sense. I've add thelogic to return `false `from `AuthorizeRequestOnChallenge()` right away if `EnableTenantDiscovery` is `false`.